src/v2/System/useFeatureFlag.tsx

```diff
@@ -37,3 +39,56 @@ export function useFeatureVariant(featureName: string): Variant | null {
 
   return variant
 }
+
+export function useTrackVariantView({
+  experiment_name,
+  variant_name,
+  payload,
+  context_owner_type,
+  context_owner_id,
+  context_owner_slug,
+}) {
+  const { trackEvent } = useTracking()
+
+  if (typeof window !== "undefined") {
+    const trackFeatureView = shouldTrack(experiment_name, variant_name)
+
+    if (trackFeatureView) {
+      trackEvent({
+        action: Schema.ActionType.ExperimentViewed,
+        service: "unleash",
+        experiment_name,
+        variant_name,
+        payload,
+        context_owner_type,
```

Comment:
@damassi - once we enforce the [cohesion schema](https://github.com/artsy/cohesion/pull/304), this is going to be required. What is the best way to grab this data?

Reply:
If i'm understanding correctly, right now this isn't a dynamic value that we can "get at" -- it's generally supplied as a static value: 

<img width="424" alt="Screen Shot 2022-03-09 at 12 57 08 PM" src="https://user-images.githubusercontent.com/236943/157534702-77c607e8-ffee-42a9-b11b-267f77a70297.png">

So if this will soon be enforced as a requirement, we might want to expand the hook arguments a bit:

```tsx
const useFeatureFlag = ({ flagName, analyticsOwnerType }) => { ... }
```

Something like that.